examples/tutorials/manage-ntp.markdown

```diff
@@ -0,0 +1,741 @@
+         classes         => results("bundle", "ntp_package");
+
+   files:
+    "$(config_file)"
+      create                => "true",
+      handle                => "ntp_files_conf",
+      perms                 => mog( "644", "root", "root" ),
+      template_method       => "inline_mustache",
+      edit_template_string  => "$(config_template_string)",
+      template_data         => mergedata( '{ "driftfile": "$(driftfile)", "servers": servers}' ),
+      classes               => results( "bundle", "ntp_config" );
+
+   services:
+     "$(ntp_service_name)" -> { "StandardsDoc 3.2.2" } 
+       service_policy => "start",
+       classes => results( "bundle", "ntp_service_running" );
+
+    ntp_config_repaired::
+     "$(ntp_service_name)" -> { "StandardsDoc 3.2.2" } 
+       service_policy => "restart",
+       classes => results( "bundle", "ntp_service_config_change" );
+
+
+   reports:
+     ntp_service_running_repaired.inform_mode::
+       "NTP service started";
+
+     ntp_service_config_change_repaired.inform_mode::
+       "NTP service restarted after configuration change";
+
+}
+```
+
+What does this policy do?
+
+Let's review the changes to the vars promises as they were the only changes made.
+
+#### vars
+
+```cf3
+bundle agent ntp
+{
+   vars:
+     linux::
+       "ntp_package_name" string => "ntp";
+       "config_file" string => "/etc/ntp.conf";
+
+       # Set the default value for driftfile
+       "driftfile"
+         string => "/var/lib/ntp/drift";
+
+       # Overwrite driftfile with value defined from Augments if it's provided
+       "driftfile"
+         string => "$(def.ntp[config][driftfile])",
+         if => isvariable( "def.ntp[config][driftfile]" );
+
+       # Set the default value for servers
+       "servers"
+         slist => { "time.nist.gov" };
+
+       # Overwrite servers with value defined from Augments if it's provided
+       "servers"
+         slist => getvalues( "def.ntp[config][servers]" ),
+         if => isvariable( "def.ntp[config][servers]" );
+```
+
+Notice two promises were introduced, one setting `driftfile` to the value of `$(def.ntp[config][driftfile])` if it is defined and one setting servers to the list of values for `def.ntp[config][servers]` if it is defined. [Augments][Augments] allows for variables to be set in the *def* bundle scope very early before policy is evaluated.
+
+You will notice that in this iteration of the variable declaration, we have removed the references to a list of IP addresses for operating system specific time servers. We've also removed the declaration of the driftfile variable. Instead, we now declare a variable called ntp_json_config_data which references JSON data files that contain this data, again classified by CentOS and Debian operating systems. The agent expects to find these JSON files in the same location as the policy, as referenced by $(this.promise_dirname). We will create the content for the JSON file further down in this tutorial.
```

Comment:
This paragraph is completely wrong. :wink:

Reply:
bah stale content from the starting source!

ack!